Bottom: 63a13b806527e5589d3dca0330e50467eb98d5b6
Top:    93f8d4382fdc0a664048c1abf011e05a6bcbaa19
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2013-10-27 16:57:09 +0000

We can just delete kicked out holes

Since we don't keep holes from broken contours, if a contour is
kicked out, we know it is destined for the bin.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 667e26d..d249a7e 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1390,7 +1390,8 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 	}
       else
 	{
-	  /* Need to check if this new hole means we need to kick out any old ones for reprocessing */
+	  /* Need to check if this new hole means we need to kick out any old
+	   * ones for deletion */
 	  while (1)
 	    {
 	      struct find_inside_info info;
@@ -1424,10 +1425,7 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 
 	      /* Remove hole from the contour */
 	      remove_contour (pa_info->pa, prev, info.result, TRUE);
-
-	      /* Add hole as the next on the list to be processed in this very function */
-	      info.result->next = *src;
-	      *src = info.result;
+	      poly_DelContour (&info.result);
 	    }
 	  /* End check for kicked out holes */
